hearth/lib/hearth/connection_pool.rb

```diff
@@ -0,0 +1,103 @@
+# frozen_string_literal: true
+
+module Hearth
+  # @api private
+  class ConnectionPool
+    @pools_mutex = Mutex.new
+    @pools = {}
+
+    class << self
+      # @return [ConnectionPool]
+      def for(config = {})
+        @pools_mutex.synchronize do
+          puts "using connection pool for #{config}\n" if @pools[config]
+          puts "creating connection pool for #{config}\n" unless @pools[config]
+          @pools[config] ||= new
+        end
+      end
+
+      # @return [Array<ConnectionPool>] Returns a list of the
+      #   constructed connection pools.
+      def pools
```

Comment:
Is there a reason to expose all the pools?

Reply:
It's used in empty_connection_pools! in v3. It's a public interface. We can start with it not included.

Reply:
Yeah, I'd argue for having it private and not exposing the pools publicly.